### PR TITLE
Adding docs around the overall multi tenant experience.

### PR DESCRIPTION
**Pull request by albertshau**
first pass at overall multi-tenancy docs.  Will need to add a lot more detail and pictures once the UI is up.

### CODE REVIEW

File: docs/docs/source/guide/admin/plugin-resources.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+.. include:: /guide/admin/admin-links.rst
+
+There is out of the box support for creating Hadoop clusters. Internally, the installation, configuration, and management of Hadoop
+services is handled on the provisioner by the Chef Automator plugin. The Chef Automator plugin uses out of the box cookbooks to 
+execute all tasks it needs to perform. However, administrators may wish to manage other sorts of clusters using other sorts of 
+services. For example, you may want to create an Elastic Search template to create an indexing and search cluster. 
+In order to allow simple and easy expansion of service and template capabilities, plugins are able to define different
+types of resources they can use to perform tasks. For example, the Chef Automator plugin defines 3 types of resources: 
+cookbooks, databags, and roles. We include several cookbooks out of the box, but the plugin resource system allows administrators to
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugin-resources.rst]: data bags

File: docs/docs/source/guide/admin/plugin-resources.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+evolve and update, new versions are uploaded and synced. If a resource ever needs to be removed completely, it is recalled, synced,
+and finally deleted. All resource operations can be performed through the UI or directly through the REST APIs documented :doc:`here </rest/plugins>`  
+
+Uploading Resources
+===================
+
+Plugin resources can only be uploaded by a tenant administrator. It can be done through the UI, or through the REST API
+documented :doc:`here </rest/plugins>`. Each time a resource is uploaded, a new version of the resource is created. Subsequent
+interactions require the version of a resource as well as the name. Archive resources are assumed to be constructed such that
+the name of the resource is a top level folder in the expanded archive. A newly uploaded resource beings in the 'inactive' state.
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugin-resources.rst]: begins?

File: docs/docs/source/guide/admin/plugin-resources.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+An inactive resource is not visible to provisioners, meaning it cannot be used. It is also  not scheduled to be used at any point
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugin-resources.rst]: also not

File: docs/docs/source/guide/admin/plugin-resources.rst (hunk 1 above)
Comment (awholegunch) [inline docs/docs/source/guide/admin/plugin-resources.rst]: "data bags"? As that is not a real word, could it be in quotes?

**Pull request merged**